browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -1718,6 +1728,90 @@ void RewardsDOMHandler::GetBalanceReport(const base::ListValue* args) {
 }
 
+void RewardsDOMHandler::OnGetMonthlyReport(
+    const uint32_t month,
+    const uint32_t year,
+    const brave_rewards::MonthlyReport& report) {
+  if (!web_ui()->CanCallJavascript()) {
+    return;
+  }
+
+  base::Value data(base::Value::Type::DICTIONARY);
+  data.SetIntKey("month", month);
+  data.SetIntKey("year", year);
+
+  base::Value balance_report(base::Value::Type::DICTIONARY);
+  balance_report.SetDoubleKey("grant", report.balance.grants);
+  balance_report.SetDoubleKey("ads", report.balance.earning_from_ads);
+  balance_report.SetDoubleKey("contribute", report.balance.auto_contribute);
+  balance_report.SetDoubleKey("donation", report.balance.recurring_donation);
+  balance_report.SetDoubleKey("tips", report.balance.one_time_donation);
+
+  base::Value transactions(base::Value::Type::LIST);
+  for (const auto& item : report.transactions) {
+    base::Value transaction_report(base::Value::Type::DICTIONARY);
+    transaction_report.SetDoubleKey("amount", item.amount);
+    transaction_report.SetIntKey("type", item.type);
+    transaction_report.SetIntKey("created_at", item.created_at);
+
+    transactions.Append(std::move(transaction_report));
+  }
+
+  base::Value contributions(base::Value::Type::LIST);
+  for (const auto& item : report.contributions) {
+    base::Value publishers(base::Value::Type::LIST);
+    for (const auto& item : item.publishers) {
+      base::Value publisher(base::Value::Type::DICTIONARY);
+      publisher.SetStringKey("id", item.id);
+      publisher.SetDoubleKey("percentage", item.percentage);
+      publisher.SetStringKey("publisherKey", item.id);
+      publisher.SetIntKey("status", item.status);
+      publisher.SetStringKey("name", item.name);
+      publisher.SetStringKey("provider", item.provider);
+      publisher.SetStringKey("url", item.url);
+      publisher.SetStringKey("favIcon", item.favicon_url);
+      publishers.Append(std::move(publisher));
+    }
+
+    base::Value contribution_report(base::Value::Type::DICTIONARY);
+    contribution_report.SetDoubleKey("amount", item.amount);
+    contribution_report.SetIntKey("type", item.type);
+    contribution_report.SetIntKey("created_at", item.created_at);
+    contribution_report.SetKey("publishers", std::move(publishers));
+    contributions.Append(std::move(contribution_report));
+  }
+
+  base::Value report_base(base::Value::Type::DICTIONARY);
+  report_base.SetKey("balance", std::move(balance_report));
+  report_base.SetKey("transactions", std::move(transactions));
+  report_base.SetKey("contributions", std::move(contributions));
+
+  data.SetKey("report", std::move(report_base));
+
+  web_ui()->CallJavascriptFunctionUnsafe(
+    "brave_rewards.monthlyReport",
+    data);
+}
+
+void RewardsDOMHandler::GetMonthlyReport(const base::ListValue* args) {
+  CHECK_EQ(2U, args->GetSize());
+  if (!rewards_service_) {
+    return;
+  }
+
+  const uint32_t month = args->GetList()[0].GetInt();
+  const uint32_t year = args->GetList()[1].GetInt();
```

Comment:
What happens if args->GetSize() != 2? the CHECK_EQ may record this but will not return/fail gracefully so these month/year dereferences would crash.

Reply:
`CHECK_EQ` so this code will not be triggered if != 2

Reply:
`CHECK_EQ` will throw an exception on release builds, should be a `DCHECK` along with defensive code as above

Reply:
please log an issue and we will do it in one go